### PR TITLE
[HowTo] How to Monitor SSL Certificates Expiry Dates for subdomains

### PR DESCRIPTION
**Pull request by unrealwork**

### CODE REVIEW

**Comment by kghamilton89** [comment]
Proofread

**Comment by rodionos** [comment]
Update PR with version where we avoid downloading and replacing files manually. Use environment variables substitutions.

**Comment by rodionos** [comment]
1. Review https://developers.google.com/style and try your best to bring the article in line with guidelines (minimize passive voice for example)
2. The sandbox now allows a simple way of configuring the default Slack webhook: https://github.com/axibase/dockers/tree/atsd-sandbox#outgoing-webhooks. Just pass the token.
3. Why do we need tutorial directory. I don't think it's necessary.
4. Check your spelling - what is 'that norify'?
5. In 'Disable Unnecessary Notifications' refer to sandbox document where default access credentials are specified.
6. What's a good way to verify that everything is working, to validate the installation?

**Comment by kghamilton89** [comment]
@rodionos Looks like we have a stray PR that fell to the wayside. This is a really great tool, it deserves publication.
1. Moved into correct directory.
2. Updated to be consistent with style guide.
3. Spelling and other lint repaired.
4. Removed **Disable Unnecessary Notifications** section, as requested.
5. `./` prefix.
6. Installation is verified by test message.

---

Travis is failing due to the letter s, the same problem as the last PR.

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# How to Monitor SSL Certificates Expiry Dates for subdomains
+
+## Overview
+
+[Axibase Collector](https://github.com/axibase/axibase-collector/#overview) provides tools to collect information about SSL certificates. The data stored in ATSD is useful to monitor SSL certificates for the subdomains of a top domain and notify a user via [Slack](https://axibase.com/docs/atsd/api/data/messages/webhook.html#slack) using web notifications triggered by the [Rule Engine](https://github.com/axibase/atsd/tree/master/rule-engine#overview) that notify when a certificate expiry date is approaching or modified. Query a list of domains using SQL in the [CRT Certificate Search](https://crt.sh) portal created by [Comodo Group](https://www.comodo.com).
```
Comment (rodionos) [inline tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: Long sentence. Split, simplify.

Comment (rodionos) [inline tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: Link to Slack is wrong.

Comment (rodionos) [inline tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: Link to atsd docs, not to github repo

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+### Tools and resources
+
+This tutorial uses the following tools:
```
Comment (rodionos) [inline tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: It's not a tutorial. This is integration guide. But remove this section as verbiage, or rephrase it, for example, "The crt.sh database, maintained by ... is a public service that allows you to obtain a list of SSL certificates issued for a top-level domain. The crt.sh Prostgres database also provides JDBC connectivity to retrieve the domains ..."

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+* [`crt.sh`](https://crt.sh): Certificate search engine to retrieve information about SSL certificates issued by Comodo Group. The portal provides open access to their database.
+* [`atsd-sandbox`](https://github.com/axibase/dockers/tree/atsd-sandbox#overview): Docker image running ATSD and Collector.
+
+## Procedure
+
+1. [Prepare job and rules for domain](#prepare-files-for-domain).
+2. [Configure and launch ATSD Sandbox](#configure-and-launch-atsd-sandbox).
+3. [Begin receiving notifications](#notifications).
```
Comment (rodionos) [inline tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: Begin receiving - verbiage

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### Prepare Files for Domain
+
+Create a directory to store tutorial files and navigate to the newly created directory from the terminal.
```
Comment (rodionos) [inline tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: This is not a tutorial. We don't say terminal, we say console.

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+```bash
+mkdir tutorial && cd ./tutorial
+```
+
+### Configure Slack Notifications
+
+ATSD Sandbox can pass a path to a file containing web notification properties as the parameters.
```
Comment (rodionos) [inline tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: Refactor due to https://github.com/axibase/dockers/tree/atsd-sandbox#outgoing-webhooks using `SLACK_TOKEN`

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+> See [Web Notifications Documentation](https://github.com/axibase/dockers/tree/atsd-sandbox#web-notifications-configuration) for more information.
+
+### Configure and Launch ATSD Sandbox
+
+Job file contains placeholder `${ENV.TOP_DOMAIN}` instead of a real domain name. Set `TOP_DOMAIN` environment variable to an actual domain name. Collector replaces the placeholder with variable name during the import process. In the launch command shown below, `TOP.DOMAIN` is set to [axibase.com](https://axibase.com).
+
+Start ATSD Sandbox with the required [environment variables](https://github.com/axibase/dockers/tree/atsd-sandbox#container-parameters):
+
+```bash
+docker run -d -p 8443:8443 -p 9443:9443 -p 8081:8081 \
```
Comment (rodionos) [inline tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: Simplify command based on latest sandbox capabilities: https://github.com/axibase/dockers/tree/atsd-sandbox

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  --name=atsd-sandbox \
+  --volume $(pwd):/import \
+  --env TOP_DOMAIN=axibase.com \
+  --env ATSD_IMPORT_PATH='https://raw.githubusercontent.com/axibase/atsd-use-cases/dev-howto-monitor-ssl-for-domains/how-to/atsd-sandbox/monitor-ssl-expiry-dates/resources/ssl-certificates-files.tar.gz' \
+  --env COLLECTOR_IMPORT_PATH='https://raw.githubusercontent.com/axibase/atsd-use-cases/dev-howto-monitor-ssl-for-domains/how-to/atsd-sandbox/monitor-ssl-expiry-dates/resources/job_http_subdomains-ssl-certificates.xml' \
+  --env SLACK_CONFIG="slack.properties" \
+  axibase/atsd-sandbox:latest
+```
+
```
Comment (rodionos) [inline tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: After command is launch, what should use check in docker -f logs to verify the correct start: e.g. job executed ok, Slack message sent etc.

**Comment by rodionos** [comment]
Most importantly, does this integration work? 

Test on a domain that has an SSL certificate that expires in less than 30 days. The rule should deliver an alert for such a subdomain when the sandbox is launched.

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md (hunk 1 above)
Comment (kghamilton89) [reply tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: Broke the sentence into two.

Comment (kghamilton89) [reply tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: Replaced to default Slack site, as it was.

Comment (kghamilton89) [reply tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: Replaced

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md (hunk 2 above)
Comment (kghamilton89) [reply tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: changed.

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md (hunk 3 above)
Comment (kghamilton89) [reply tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: changed to receive

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md (hunk 4 above)
Comment (kghamilton89) [reply tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: changed.

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md (hunk 5 above)
Comment (kghamilton89) [reply tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: variable names updated

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md (hunk 6 above)
Comment (kghamilton89) [reply tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: I believe launch command is already using most current/simple syntax

File: tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md (hunk 7 above)
Comment (kghamilton89) [reply tutorials/atsd-sandbox/monitor-ssl-expiry-dates/README.md]: @unrealwork

**Pull request merged**